addon/-private/deep-tracked.ts

```diff
@@ -134,11 +136,15 @@ const arrayProxyHandler: ProxyHandler<object> = {
         readCollection(target);
         readStorage(target, parsed);
 
-        return deepTracked(Reflect.get(target, parsed, receiver));
+        return deepTracked(value);
       }
-    }
 
-    let value = Reflect.get(target, property, receiver);
+      if (ARRAY_COLLECTION_PROPERTIES.includes(property)) {
```

Comment:
@gossi this is the fix -- any chance you can test this with the code where you found the issue? want to make sure this approach isn't _too specific_